sdk/search/azure-search/src/main/java/com/azure/search/models/DataDeletionDetectionPolicy.java

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
+/**
+ * Abstract base class for data deletion detection policies.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "@odata.type", defaultImpl = DataDeletionDetectionPolicy.class)
+@JsonTypeName("DataDeletionDetectionPolicy")
+@JsonSubTypes({
+    @JsonSubTypes.Type(name = "#Microsoft.Azure.Search.SoftDeleteColumnDeletionDetectionPolicy", value = SoftDeleteColumnDeletionDetectionPolicy.class)
```

Comment:
We have another deletion detection policy, buts its not in the swagger spec. NativeBlobSoftDeleteDeletionDetectionPolicy. I'll add a work item to update the spec